src/Albany_Application.cpp

```diff
@@ -442,7 +413,7 @@ Application::initialSetUp(const RCP<Teuchos::ParameterList>& params)
   // Schwarz problems.
   countScale = 0;
   // Create discretization object
-  discFactory = rcp(new Albany::DiscretizationFactory(params, comm, expl));
+  discFactory = rcp(new Albany::DiscretizationFactory(params, comm, num_params_, expl));
```

Comment:
you could deduce the number of parameters from params.

Reply:
I agree, but there's a routine that does this from the PL in Albany::Application, and I didn't want to duplicate that code in the discretization factory; so I just pass it in.  I don't think it's a big deal since we only construct the discretization factory once.

Reply:
We can take that code and put it in Albany_Utils.hpp. Having the num_params_ extra arg is somewhat suggesting it is an independent argument, while in fact it must agree with what is in the parameter list.

Reply:
@bartgol : I like this suggestion.  Let me try it.